depends/libhdfs3/src/client/FileEncryptionInfo.h

```diff
@@ -0,0 +1,93 @@
+/********************************************************************
+ * 2014 -
+ * open source under Apache License Version 2.0
+ ********************************************************************/
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef _HDFS_LIBHDFS3_CLIENT_FILEENCRYPTIONINFO_H_
+#define _HDFS_LIBHDFS3_CLIENT_FILEENCRYPTIONINFO_H_
+
+#include <string>
+
+namespace Hdfs {
+
+class FileEncryptionInfo {
+public:
+    FileEncryptionInfo() : 
+		cryptoProtocolVersion(0), suite(0){
+    }
+
+    int getSuite() const {
+        return suite;
+    }
+
+    void setSuite(int suite) {
+        this->suite = suite;
+    }
+
+    int getCryptoProtocolVersion() const {
+        return cryptoProtocolVersion;
+    }
+
+    void setCryptoProtocolVersion(int cryptoProtocolVersion) {
+        this->cryptoProtocolVersion = cryptoProtocolVersion;
+    }
+
+    const char * getKey() const{
+        return key.c_str();
```

Comment:
According to libhdfs3 coding format, all the related class choose char * instead of sting as the data type. In the meanwhile, ConstructHdfsEncryptionFileInfo() will call Strdup() to memcpy char * in Hdfs.cpp. So, will use char * data type. Thank you, interma.